File: server/routers/airports.py
from fastapi import APIRouter, Depends
from typing import List
from sqlalchemy.orm import Session
from database import get_db
from models.airport import Airport

router = APIRouter()


@router.get(
    "",
    tags=["Sân bay"],
    name="Lấy danh sách sân bay",
    description="Lấy danh sách tất cả sân bay",
    response_model=List[dict],
)
def get_airports(db: Session = Depends(get_db)):
    """Lấy danh sách tất cả sân bay"""
    airports = db.query(Airport).all()
    return [
        {
            "id": airport.id,
            "name": airport.name,
            "city": airport.city,
            "display": f"{airport.city} ({airport.id}) - {airport.name}"
        }
        for airport in airports
    ]


@router.get(
    "/search",
    tags=["Sân bay"],
    name="Tìm kiếm sân bay",
    description="Tìm kiếm sân bay theo tên thành phố hoặc tên sân bay",
    response_model=List[dict],
)
def search_airports(q: str, db: Session = Depends(get_db)):
    """Tìm kiếm sân bay theo query string"""
    airports = db.query(Airport).filter(
        Airport.name.ilike(f"%{q}%") | 
        Airport.city.ilike(f"%{q}%") |
        Airport.id.ilike(f"%{q}%")
    ).all()
    
    return [
        {
            "id": airport.id,
            "name": airport.name,
            "city": airport.city,
            "display": f"{airport.city} ({airport.id}) - {airport.name}"
        }
        for airport in airports
    ]

File: web/src/apis/airports.ts
import axiosInstance from '../lib/axios'

export interface Airport {
  id: string
  name: string
  city: string
  display: string
}

export const airportApi = {
  // Lấy tất cả sân bay
  getAll: async (): Promise<Airport[]> => {
    const response = await axiosInstance.get('/airports')
    return response.data
  },

  // Tìm kiếm sân bay
  search: async (query: string): Promise<Airport[]> => {
    const response = await axiosInstance.get(`/airports/search?q=${encodeURIComponent(query)}`)
    return response.data
  }
}
